### PR TITLE
Implements `opened` attribute in Socket

### PR DESCRIPTION
**Pull request by dom96**
This was implemented as part of another PR, all this PR does is expose it as an attribute in a Socket instance.

Defined here in the sockets spec https://sockets-api.proposal.wintercg.org/#opened-attribute.

### CODE REVIEW

**Comment by jasnell** [comment]
Does this include resolving the `opened` promise with the new `SocketInfo`  structure?

**Comment by dom96** [comment]
Fair point, it does not. Will add that.

**Review by vickykont — APPROVED** [review]
Looks good to me! Leaving it up to you to decide whether you should wait for one more review from someone with more context.

P.S. I understand that `alpn` is intentionally missing.

**Pull request merged**
